### PR TITLE
Mailchimp

### PR DESCRIPTION
**Pull request by Greg-Boggs**
The form still requires theming. But this is the implementation for a sign up form based on a Sign Up form block with an easy to configure interface and some helpful error checking to get folks started.

### CODE REVIEW

File: modules/bene_core/src/Plugin/Block/NewsletterSignupBlock.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use Drupal\Core\Block\BlockBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
+use \Drupal\mailchimp_signup\Form\MailchimpSignupPageForm;
```
Comment (tauno) [inline modules/bene_core/src/Plugin/Block/NewsletterSignupBlock.php]: Drop the leading `\`. PHPCS is complaining about it and it's not needed.

File: modules/bene_core/src/Plugin/Block/NewsletterSignupBlock.php [hunk 2]
```diff
@@ -79,6 +80,60 @@ public function blockForm($form, FormStateInterface $form_state) {
...
+      '#type' => 'fieldset',
+      '#title' => $this->t('MailChimp settings'),
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[style]"]' => ['value' => 'embedded'],
+        ],
+      ],
+    ];
+
+    // Help the user turn on MailChimp
```
Comment (tauno) [inline modules/bene_core/src/Plugin/Block/NewsletterSignupBlock.php]: Add `.` at the end to keep PHPCS happy.

**Review by tauno — CHANGES_REQUESTED** [review]
Before configuring a Mailchimp key, if you select Mailchimp on the bene newsletter block and save the form you get an error page. This is without the mailchimp module enabled. We should catch this case and fail validation if the admin tries to select Mailchimp without properly configuring a mailchimp form. Note that all pages then result in errors due to the block failing to load the mailchimp form. 

I added the mailchimp module as a dependency in the composer file for the profile.

**Comment by tauno** [comment]
@Greg-Boggs heads up that I did rebase and force push to get these changes applied on top of the core security updates from last week

**Comment by Greg-Boggs** [comment]
@tauno I have added improvements to the error checking to prevent the fatal error. I attempted to make the signup block field required, but this does not work because hidden required fields prevent the block from being saved. So, currently, I let the user have a signup block with no form.

**Comment by tauno** [comment]
Tested by deleting the currently configured signup form. After clearing the cache, the block is causing an error again.

There are also a couple of minor code style fixes in my review above.

**Comment by tauno** [comment]
@Greg-Boggs further - I can no longer save the newsletter block configuration when mailchimp is enabled but there are no lists configured. I'm guessing this is related to the required field issue you mentioned. If we can make the field required with the form api then we should check for a value during validation - when MailChimp is the selected option.

**Comment by Greg-Boggs** [comment]
I got the two coding standard fixes in now. Please let me know if you see any others. PHPCS is failing on me, and I haven't had time to sort out why. 

I've also now included an error check to cover the case where a Signup Block is created, assigned, displayed and then later deleted. If you refresh the page twice without creating a sign up block, the page shouldn't display any errors.

**Comment by Greg-Boggs** [comment]
I have removed the #required = TRUE until we have a better solution, and I’ve created a branch with my latest attempt.

https://github.com/thinkshout/bene/commit/cf2b1e54b2578143aca2e9220461f4deff3d7895

I’m going to pause because I can’t get this to work quite right for the case of a signup form is created, then the signup block is deleted because the form still has the previous signup block in form state.

To complete it, we need set a default value such as --- and then check to see if the value is set to ---.

I believe this only comes up in the case where you create a sign up list, set it, then delete it.

**Comment by tauno** [comment]
Merged a different branch in with the same base changes.

**Pull request closed without merging**